lib/board.rb

```diff
@@ -1,2 +1,62 @@
 class Board
+  #1) how to populate gird with ships
+  #2) need to display board and update
+
+  attr_accessor :grid
+
+  def initialize(grid = Board.default_grid)
+    @grid = grid
+  end
+
+  def self.default_grid
+    Array.new(10) { Array.new(10)}
+  end
+
+  def [](pos)
+    row, col = pos
+    grid[row][col]
+  end
+
+  def over?
+    grid.flatten.none? { |pos| pos.nil? }
+  end
+
+  def populate_grid
+#how many ships do we populate with? 5
+#create random size
+#how to randomly place on grid, mark with :s
+#need help
+    #rand_ship_size = rand(1..grid.length)
+    #rand_ship_start = [rand(1..grid.length), rand(1..grid.length)
+    #5.times do |ship|
+      #ship = grid[rand_ship_start] == :s if grid[rand_start.empty?]
+    #end
+  end
+
+  def count
+    grid.flatten.select { |pos| pos == :s }.length
+  end
+
+  def empty?(pos=nil) # why do we make the default value nil?
```

Comment:
This allows the method to be called without any parameter:

```ruby
board.empty?([0,0])
# true or false
board.empty?
# true or false
```

Reply:
This is related to the spec at https://github.com/hvanhonacker/battleship-project/blob/master/spec/board_spec.rb#L55

Reply:
ahhh okay